packages/react-server/src/ReactFlightServer.js

```diff
@@ -810,12 +822,36 @@ function serializeMap(
   request: Request,
   map: Map<ReactClientValue, ReactClientValue>,
 ): string {
```

Comment:
Maps with object keys (or Sets of objects) seems like an interesting case where the object identity is virtually guaranteed to be semantic. Should we even allow that if there's no guarantee on it matching other places where that object is sent down (including other Map/Set instances)?